examples/simpleGov/x/simple_governance/handler.go

```diff
@@ -0,0 +1,173 @@
+package simpleGovernance
+
+import (
+	"encoding/binary"
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+	abci "github.com/tendermint/abci/types"
+)
+
+// Minimum proposal deposit
+var minDeposit = sdk.NewInt(int64(100))
+
+const votingPeriod = 1209600
+
+func int64ToBytes(i int64) []byte {
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, uint64(i))
+	return b
+}
+
+// NewHandler creates a new handler for all simple_gov type messages.
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case SubmitProposalMsg:
+			return handleSubmitProposalMsg(ctx, k, msg)
+		case VoteMsg:
+			return handleVoteMsg(ctx, k, msg)
+		default:
+			errMsg := "Unrecognized gov Msg type: " + reflect.TypeOf(msg).Name()
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// NewEndBlocker checks proposals and generates a EndBlocker
+func NewEndBlocker(k Keeper) sdk.EndBlocker {
+	return func(ctx sdk.Context, req abci.RequestEndBlock) (res abci.ResponseEndBlock) {
+		err := checkProposal(ctx, k)
+		if err != nil {
+			panic(err)
+		}
+		return
+	}
+}
+
+// checkProposal checks if the proposal reached the end of the voting period
+// and handles the logic of closing it
+func checkProposal(ctx sdk.Context, k Keeper) sdk.Error {
+	proposal, err := k.ProposalQueueHead(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Proposal reached the end of the voting period
+	if ctx.BlockHeight() >= proposal.SubmitBlock+votingPeriod && proposal.IsOpen() {
+		k.ProposalQueuePop(ctx)
+
+		nonAbstainTotal := proposal.YesVotes + proposal.NoVotes
+		if float64(proposal.YesVotes)/float64(nonAbstainTotal) > float64(0.5) { // TODO: Deal with decimals
+
+			// Refund deposit
+			_, _, err := k.ck.AddCoins(ctx, proposal.Submitter, proposal.Deposit)
+			if err != nil {
+				return err
+			}
+			proposal.State = "Accepted"
```

Comment:
Doesn't this proposal have to be serialized and persisted back into the key-value store after the `State` has been changed?